packages/components/src/menu-item/test/index.js

```diff
@@ -107,4 +107,40 @@ describe( 'MenuItem', () => {
 		expect( checkboxMenuItem ).toBeChecked();
 		expect( checkboxMenuItem ).toHaveAttribute( 'aria-checked', 'true' );
 	} );
+
+	it( 'should not render shortcut or right icon if suffix provided', () => {
+		render(
+			<MenuItem
+				icon="Icon"
+				iconPosition="right"
+				role="menuitemcheckbox"
+				shortcut="Shortcut"
+				suffix="Suffix"
+			>
+				My item
+			</MenuItem>
+		);
+
+		expect( screen.getByText( 'Suffix' ) ).toBeInTheDocument();
+		expect( screen.queryByText( 'Shortcut' ) ).not.toBeInTheDocument();
+		expect( screen.queryByText( 'Icon' ) ).not.toBeInTheDocument();
```

Comment:
I don't think that the `Icon` text would ever be displayed in the component, regardless of the `suffix` prop being defined or not — in this case, I believe that the component would render something like 

```html
<span class="dashicon dashicons dashicons-Icon"></span>
```

Did you mean to pass `icon={ <span>Icon</span> }` like in the test below?

Reply:
🤦 I did mean to. Thanks. I've fixed this test in https://github.com/WordPress/gutenberg/pull/44260/commits/beba370b36be0c61949d7495bb4231d133549e61